// @canva/design/lib/esm/sdk/design/fake/fake_design_interaction_client.js
export class FakeDesignInteractionClient {
    async getDesignToken() {
        await this.delay();
        return {
            token: 'token'
        };
    }
    async getDesignMetadata() {
        await this.delay();
        return {
            title: 'title',
            defaultPageDimensions: {
                width: 800,
                height: 600
            },
            pageMetadata: [
                {
                    type: 'absolute',
                    dimensions: {
                        width: 800,
                        height: 600
                    },
                    title: 'page1'
                },
                {
                    type: 'unsupported'
                },
                {
                    type: 'absolute',
                    dimensions: {
                        width: 300,
                        height: 900
                    },
                    title: undefined
                }
            ],
            durationInSeconds: 30
        };
    }
    async getDesignTemplateMetadata() {
        await this.delay();
        return {
            templates: [
                {
                    keywords: [
                        'word1',
                        'word2'
                    ],
                    domain: 'brand'
                }
            ]
        };
    }
    async addNativeElement(element) {
        await this.delay();
    }
    async addElementAtPoint(element) {
        await this.delay();
    }
    async addElementAtCursor(element) {
        await this.delay();
    }
    async addAudioTrack(audioTrack) {
        await this.delay();
    }
    async addPage(opts) {
        await this.delay();
    }
    async setCurrentPageBackground(opts) {
        await this.delay();
    }
    async setAppElementData(appElementData, placement) {
        await this.delay();
    }
    onAppElementChange(handler) {}
    async addAppElement(opts) {
        await this.delay();
    }
    async updateAppElement(opts) {
        await this.delay();
    }
    onAppElementDrop(handler) {}
    onAppElementCreateFromPreset(handler) {}
    registerRenderAppElement(renderAppElement) {}
    async getCurrentPageContext() {
        await this.delay();
        return {
            dimensions: fakePageDimensions
        };
    }
    async getDefaultPageDimensions() {
        await this.delay();
        return fakePageDimensions;
    }
    initAppElement(appElementConfig) {
        return {
            addOrUpdateElement: async ()=>{
                await this.delay();
            },
            registerOnElementChange: (handler)=>{},
            addElement: async (opts)=>{
                await this.delay();
            }
        };
    }
    createRichtextRange() {
        return fakeRichtextRange;
    }
    async editContent(options, callback) {
        await this.delay();
        await callback({
            contents: [],
            sync: async ()=>{
                await this.delay();
            }
        });
    }
    async openDesign(options, callback) {
        await this.delay();
        await callback({
            page: fakePage,
            sync: async ()=>{
                await this.delay();
            },
            helpers: fakeOpenDesignHelpers
        });
    }
    constructor(delay){
        this.delay = delay;
        this.selection = {
            registerOnChange: (e)=>{
                return ()=>{};
            }
        };
        this.overlay = {
            registerOnCanOpen: (e)=>{
                return ()=>{};
            }
        };
        this.publish = {
            launch: ()=>this.delay()
        };
        this.bulkCreate = {
            launch: ()=>this.delay()
        };
    }
}
const fakeBounds = {
    index: 0,
    length: 12
};
const fakeTextRegion = {
    text: 'Hello World!',
    formatting: {
        fontWeight: 'bold'
    }
};
const fakeRichtextRange = {
    formatParagraph: ()=>{},
    formatText: ()=>{},
    appendText: ()=>{
        return {
            bounds: fakeBounds
        };
    },
    replaceText: ()=>{
        return {
            bounds: fakeBounds
        };
    },
    readPlaintext: ()=>'',
    readTextRegions: ()=>[
            fakeTextRegion
        ]
};
const fakePageDimensions = {
    width: 800,
    height: 600
};
const fakeFill = {
    mediaContainer: {
        ref: undefined,
        set (state) {}
    },
    colorContainer: {
        ref: {
            type: 'solid',
            color: '#000000'
        },
        set (state) {}
    }
};
const fakeColorContainer = {
    ref: {
        type: 'solid',
        color: '#000000'
    },
    set (state) {}
};
const fakeElementList = {
    forEach (callback) {},
    delete (item) {},
    moveAfter (ref, item) {},
    moveBefore (ref, item) {},
    toArray () {
        return [
            fakeRectElement,
            fakeShapeElement
        ];
    },
    count () {
        return 0;
    },
    filter (filter) {},
    insertAfter (ref, state) {
        return fakeRectElement;
    },
    insertBefore (ref, state) {
        return fakeRectElement;
    }
};
export const fakePage = {
    type: 'absolute',
    locked: false,
    dimensions: fakePageDimensions,
    background: fakeFill,
    elements: fakeElementList
};
const fakeElement = {
    top: 0,
    left: 0,
    rotation: 0,
    transparency: 0,
    locked: true,
    width: 100,
    height: 100
};
const fakeRectElement = {
    ...fakeElement,
    type: 'rect',
    fill: fakeFill,
    stroke: {
        weight: 10,
        colorContainer: fakeColorContainer
    }
};
const fakeShapeElement = {
    ...fakeElement,
    type: 'shape',
    viewBox: {
        top: 0,
        left: 0,
        width: 100,
        height: 100
    },
    paths: {
        count: ()=>0,
        toArray: ()=>[],
        forEach: (callback)=>{},
        filter: (filter)=>[]
    }
};
const fakeEmbedElement = {
    ...fakeElement,
    type: 'embed',
    url: 'https://canva.dev/docs/apps/testing/'
};
const fakeTextElement = {
    ...fakeElement,
    type: 'text',
    text: fakeRichtextRange
};
const fakeElementStateBuilder = {
    createEmbedElement (opts) {
        return fakeEmbedElement;
    },
    createRectElement (opts) {
        return {
            ...fakeRectElement,
            fill: {
                colorContainer: {
                    type: 'solid',
                    color: '#000000'
                },
                mediaContainer: undefined
            },
            stroke: {
                weight: 10,
                colorContainer: {
                    type: 'solid',
                    color: '#000000'
                }
            }
        };
    },
    createRichtextRange () {
        return fakeRichtextRange;
    },
    createShapeElement (opts) {
        return {
            ...fakeShapeElement,
            paths: []
        };
    },
    createTextElement (opts) {
        return {
            ...fakeTextElement,
            text: {
                regions: []
            }
        };
    }
};
const fakeAsyncOperations = {
    async group () {
        const fakeGroupElement = {
            top: 0,
            left: 0,
            width: 100,
            height: 100,
            locked: false,
            rotation: 0,
            transparency: 0,
            type: 'group',
            contents: {
                count: ()=>0,
                toArray: ()=>[],
                forEach: (callback)=>{},
                filter: (filter)=>[]
            }
        };
        return fakeGroupElement;
    },
    async ungroup () {
        const fakeRectElement = {
            top: 0,
            left: 0,
            width: 100,
            height: 100,
            locked: false,
            rotation: 0,
            transparency: 0,
            type: 'rect',
            fill: {
                mediaContainer: {
                    set () {},
                    ref: undefined
                },
                colorContainer: {
                    set () {},
                    ref: undefined
                }
            },
            stroke: {
                weight: 10,
                colorContainer: {
                    set () {},
                    ref: {
                        type: 'solid',
                        color: '#000000'
                    }
                }
            }
        };
        return [
            fakeRectElement
        ];
    }
};
export const fakeOpenDesignHelpers = {
    elementStateBuilder: fakeElementStateBuilder,
    group: fakeAsyncOperations.group,
    ungroup: fakeAsyncOperations.ungroup,
    snapshot: ()=>{
        return {};
    }
};
